src/GitVersion.Core/Extensions/StringExtensions.cs

```diff
@@ -130,9 +131,15 @@ public static string RegexReplace(this string input, string pattern, string repl
             return Regex.Replace(input, pattern, replace, options);
         }
 
-        public static bool IsEquivalentTo(this string self, string other)
+        public static bool IsEquivalentTo(this string self, string? other)
         {
             return string.Equals(self, other, StringComparison.OrdinalIgnoreCase);
         }
+
+        /// <inheritdoc cref="StringExtensions.IsNullOrEmpty(string)"/>
+        public static bool IsNullOrEmpty([NotNullWhen(returnValue: false)] string? value) => string.IsNullOrEmpty(value);
```

Comment:
Simple wrappers with some annotation to give hints to the analyzers/compiler about nullability states.

Let me know if you want these somewhere else.

Reply:
Can we make these into proper extension methods?

```suggestion
        public static bool IsNullOrEmpty([NotNullWhen(returnValue: false)] this string? value) => string.IsNullOrEmpty(value);
```

Reply:
Absolutely! I wasn't sure what was your preference.

FYI, in roslyn they went with `RoslynString.XXX`.

Reply:
Done.